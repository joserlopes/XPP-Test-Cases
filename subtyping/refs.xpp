type struct1 = struct { #a: int };
let a: ref<struct1> = box({ #a = 10 });
a;;

type struct1 = struct { #a: int };
let a: ref<struct1> = box({ #a = 10, #b = 10 });
a;;

type struct1 = struct { #a: int, #b: int };
let a: ref<struct1> = box({ #a = 10 });
a;;

type struct1 = struct { #a: int, #b: int };
let a: ref<struct1> = box({ #a = 10, #b = 30 });
a;;
