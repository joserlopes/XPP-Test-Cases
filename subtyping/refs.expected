Value: 1
Type: int

Value: 1
Type: int

Value: 1
Type: int

Value: 30
Type: int

Value: 3
Type: int
